### PR TITLE
Week 19 - Rasmus Persson

### PR DESCRIPTION
**Pull request by rasmuselmgren**
Live Project: https://week19-project-happy-thought.herokuapp.com/

### CODE REVIEW

**Review by CarolineF3 — APPROVED** [review]
Very clean and readable code, well done!✨ It's nice that you're trying out some new stuff (the exec function and the personalized/customised res.send message).  You might wanna add some documentation in ReadMe to complete your project.

File: server.js [hunk 1]
```diff
@@ -1,30 +1,94 @@
...
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
+
+const schema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 100,
```
Comment (KarolCislowski) [inline server.js:29]: instruction specified max 140 characters messages.

File: server.js [hunk 2]
```diff
@@ -1,30 +1,94 @@
...
+  } catch (error) {
+    res.status(400).json({ error: "Failed to send post." });
+  }
+});
+
+app.post("/thoughts/:id/like", async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const likes = await Thoughts.updateOne({ _id: id }, { $inc: { heart: 1 } });
```
Comment (KarolCislowski) [inline server.js:83]: 👍🏻

File: server.js [hunk 3]
```diff
@@ -1,30 +1,94 @@
...
+  try {
+    const thoughts = await Thoughts.find()
+      .sort({ createdAt: "desc" })
+      .limit(20)
+      .exec();
+    res.json(thoughts);
+  } catch (error) {
+    res.status(400).json({
+      message: "Could not fetch the list of thoughts.",
+      error: error.errors,
```
Comment (KarolCislowski) [inline server.js:65]: nice error handling 👍🏻

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Code looks really good, but:
- Please change maxlength value to 140 characters to meet requirements;
- Please provide a link to the hosted frontend connected to this API in your Readme file.
- Present your app in the readme file in a few lines. It is very important.

**Comment by rrpersson** [comment]
I've changed the code and ReadMe. I don't have it connectd to a frontend since that was not part of the assignment.

**Review by KarolCislowski — APPROVED** [review]
Nice App

**Pull request closed without merging**